lib/triangle_farms/calendar/event_api/client.ex

```diff
@@ -59,8 +59,8 @@ defmodule TriangleFarms.Calendar.EventAPI.Client do
   end
 
   defp to_event(event = %{"summary" => summary, "description" => desc, "location" => loc}) do
-    start_at = event |> get_in(["start", "dateTime"]) |> Timex.parse!("{RFC3339}")
-    end_at = event |> get_in(["end", "dateTime"]) |> Timex.parse!("{RFC3339}")
+    start_at = event |> get_in(["start", "dateTime"]) |> Time.parse
```

Comment:
Something that might be interesting to explore. You can use Ecto without defining a schema ([explained by Plataformatec here and elsewhere](http://blog.plataformatec.com.br/2016/05/ectos-insert_all-and-schemaless-queries/).) You can also defined custom Ecto Types that allows for their own casting. This function could them remap the received map and pass it along to a function on Event that will return your struct and handle the date time parsing in it's cast. Just an idea. Don't know if it's worth the effort.

Reply:
I think this would be worth it if the app got more complex. It's gonna stay simple for now so I'll probably punt on that. I do like the idea of moving some of this junk to the wrapper Time module though.